MvvmCross/Binding/iOS/Target/MvxUIControlTargetBinding.cs

```diff
@@ -0,0 +1,209 @@
+
+    public class MvxUIControlTargetBinding : MvxConvertingTargetBinding
+    {
+        private ICommand _command;
+        private IDisposable _canExecuteSubscription;
+        private readonly EventHandler<EventArgs> _canExecuteEventHandler;
+		private readonly string _controlEvent;
+        protected UIControl Control => base.Target as UIControl;
+
+		public MvxUIControlTargetBinding(UIControl control, string controlEvent)
+            : base(control)
+        {
+			_controlEvent = controlEvent;
+
+            if (control == null)
+            {
+                MvxBindingTrace.Trace(MvxTraceLevel.Error, "Error - UIControl is null in MvxUIControlTargetBinding");
+            }
+            else
+            {
+				AddHandler(control);	
+            }
+
+            this._canExecuteEventHandler = new EventHandler<EventArgs>(this.OnCanExecuteChanged);
+        }
+
+        private void ControlEvent(object sender, EventArgs eventArgs)
+        {
+            if (this._command == null)
+                return;
+
+            if (!this._command.CanExecute(null))
+                return;
+
+            this._command.Execute(null);
+        }
+
+        public override MvxBindingMode DefaultMode => MvxBindingMode.OneWay;
+
+        public override System.Type TargetType => typeof(ICommand);
+
+        protected override void SetValueImpl(object target, object value)
+        {
+            if (this._canExecuteSubscription != null)
+            {
+                this._canExecuteSubscription.Dispose();
+                this._canExecuteSubscription = null;
+            }
+            this._command = value as ICommand;
+            if (this._command != null)
+            {
+                this._canExecuteSubscription = this._command.WeakSubscribe(this._canExecuteEventHandler);
+            }
+            this.RefreshEnabledState();
+        }
+
+        private void RefreshEnabledState()
+        {
+            var view = this.Control;
+            if (view == null)
+                return;
+
+            var shouldBeEnabled = false;
+            if (this._command != null)
+            {
+                shouldBeEnabled = this._command.CanExecute(null);
+            }
+            view.Enabled = shouldBeEnabled;
+        }
+
+        private void OnCanExecuteChanged(object sender, EventArgs e)
+        {
+            this.RefreshEnabledState();
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            if (isDisposing)
+            {
+                var view = this.Control;
+                if (view != null)
+                {
+					RemoveHandler(view);
+                }
+                if (this._canExecuteSubscription != null)
+                {
+                    this._canExecuteSubscription.Dispose();
+                    this._canExecuteSubscription = null;
+                }
+            }
+            base.Dispose(isDisposing);
+        }
+
+		private void AddHandler(UIControl control)
```

Comment:
I just used the ones that were implemented before. I'm also not 100% sure if WeakEvents will use here as we attach to an existing event handler and we don't replace one.